torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -825,8 +825,6 @@ Tensor prod_backward(
   Tensor zero_idx = (input == 0).nonzero();
   if (zero_idx.sym_numel() == 0) {
     return grad * (result / input).conj();
-  } else if (zero_idx.size(0) > 1) {
```

Comment:
(!at::GradMode::is_enabled() && zero_idx.size(0) > 1)

Reply:
Thanks for the tip, @albanD. I've updated the code.